pkg/monitor/monitor/template.go

```diff
@@ -449,73 +473,67 @@ func isDMJob(jobName string) bool {
 	return false
 }
 
-func addAlertManagerUrl(pc *config.Config, cmodel *MonitorConfigModel) {
-	pc.AlertingConfig = config.AlertingConfig{
-		AlertmanagerConfigs: []*config.AlertmanagerConfig{
+func addAlertManagerUrl(cfg yaml.MapSlice, cmodel *MonitorConfigModel) yaml.MapSlice {
+	cfg = append(cfg, yaml.MapItem{
+		Key: "alerting",
+		Value: yaml.MapSlice{
 			{
-				ServiceDiscoveryConfig: config.ServiceDiscoveryConfig{
-					StaticConfigs: []*config.TargetGroup{
+				Key: "alertmanagers",
+				Value: []yaml.MapSlice{
+					{
 						{
-							Targets: []model.LabelSet{
-								{model.AddressLabel: model.LabelValue(cmodel.AlertmanagerURL)},
+							Key: "static_configs", Value: []yaml.MapSlice{
+								{
+									{
+										Key:   "targets",
+										Value: []string{cmodel.AlertmanagerURL},
+									},
+								},
 							},
 						},
 					},
 				},
 			},
 		},
-	}
+	})
+	return cfg
 }
 
-func RenderPrometheusConfig(model *MonitorConfigModel) (string, error) {
-	pc := newPrometheusConfig(model)
+func RenderPrometheusConfig(model *MonitorConfigModel) (yaml.MapSlice, error) {
+	cfg := newPrometheusConfig(model)
 	if len(model.AlertmanagerURL) > 0 {
-		addAlertManagerUrl(pc, model)
-		pc.RuleFiles = []string{
-			"/prometheus-rules/rules/*.rules.yml",
-		}
-	} else if model.EnableAlertRules {
-		// Add alert rules when `EnableAlertRules` enabled even if AlertManager not configured.
-		pc.RuleFiles = []string{
-			"/prometheus-rules/rules/*.rules.yml",
-		}
-
+		cfg = addAlertManagerUrl(cfg, model)
+	}
+	rulesPath := []string{
```

Comment:
It seems like that the logic is different as before, what if `model.EnableAlertRules` is false?